### PR TITLE
Clarify SLSA requirements.

### PR DESCRIPTION
**Pull request by MarkLodato**
Changes to requirements:
- Remove "Source Integrity", add immutable references to "Hermetic".
- Drop "Common" from SLSA 2 because it is likely expensive.

Clarifications:
- Split out "Ephemeral Environment" from "Isolation" (from #52).
- Explain that GH-generated merge commits meet Verified History (from #52).
- Clarify that all artifact references are immutable (from #52).
- Rename "Dependencies" to "Dependencies Complete" to avoid confusion.
- Define "SLSA level", "provenance", and "top-level source."
- Other minor cleanups.

### CODE REVIEW

**Review by dlorenc — APPROVED** [review]

**Comment by dlorenc** [comment]
LGTM. I'm not sure I fully understand what it means for 2/3 to be provisional though.

**Review by inferno-chromium — APPROVED** [review]
Thanks a lot!

provisional can be changed to * and can explain this * in more detail below as provisional and that will be iterating on it with feedback from community.

**Comment by inferno-chromium** [comment]
> Thanks a lot!
> 
> provisional can be changed to * and can explain this * in more detail below as provisional and that will be iterating on it with feedback from community.

it can be even done in a generic way at bottom, as lets say 1.5 can be improved as we implement demo. so provisional should allow changes to all levels if needed in future.

**Comment by inferno-chromium** [comment]
since you are here, can you rephrase "Our draft proposal is called SLSA", we shouldnt call it draft now ?

**Comment by MarkLodato** [comment]
Let me revert the "provisional" and removal of "proposed" and instead do that in a separate PR, because that is kind of a loose thread that we can keep uplling.

**Comment by MarkLodato** [comment]
Sent #55 for address the "provisional" and "proposed" aspect.
